src/main/java/org/jenkinsci/maven/plugins/hpi/TestInsertionMojo.java

```diff
@@ -102,22 +105,29 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             getLog().info("Skipping auto-test generation because we are targeting Jenkins "+target+" (at least 1.327 is required).");
             return;
         }
+        if (javaLevel != null && javaLevel.compareTo("7") < 0) {
```

Comment:
This will break on Java 10 BTW; use a numeric comparison.

Reply:
done